### PR TITLE
Add design docs for protocol sharing

### PR DESCRIPTION
**Pull request by abuccts**
Add design docs for OpenPAI protocol sharing in marketplace.

### CODE REVIEW

File: docs/protocol/marketplace.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+Protocol
+--------
+
+OpenPAI protocol is a set of schema which defines all required or optional aspects of an OpenPAI job. It has five types of components: _JOB_, _DATA_, _SCRIPT_, _DOCKERIMAGE_, and _OUTPUT_. The _JOB_ type may require other types which are prerequisite assets for an AI job.
+
+The protocol for a specific job can be written as a YAML file, which is called protocol file. A protocol file is valid if and only if it meets the schema in protocol.
+
+To implement the protocol, we need to define the schema for job and validate a YAML file for the pre-defined schema.
+
```
Comment (scarlett2018) [inline docs/protocol/marketplace.md]: define or extend? We have a base PAI protocol, right?

File: docs/protocol/marketplace.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+Repository
+----------
+
+OpenPAI repository is a directory on file system, which contains other reposirory and __valid__ protocol files.
+The structure of a repository can be defined by users, through component type, usage, or anything else.
```
Comment (scarlett2018) [inline docs/protocol/marketplace.md]: what kind of "other repository" is supported?

File: docs/protocol/marketplace.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+The repository supports file operation for protocol files or directories, and content based search among any aspect in selected components.
+
+The root directory of a repository should contain a dot file or dot directory, which records configuration or information of the repository, and indicates that it is a repository. Possible configuration and information could be the repository's protocol files list (to track all files), components content index (to search content easily), information may be needed in the future like version and access details.
+
+To implement the repository, we need to initialize the dot file or directory and update it everytime the repository is accessed.
+
```
Comment (scarlett2018) [inline docs/protocol/marketplace.md]: Can we draw an architecture or component diagram to illustrate the relationship between protocol, job files, repositories and marketplace?

File: docs/protocol/marketplace.md (hunk 1 above)
Comment (abuccts) [reply docs/protocol/marketplace.md]: Right. This part has been implemented and it refers to existing [protocol](https://github.com/Microsoft/pai/blob/pai-proto/docs/pai-protocol.yaml).

File: docs/protocol/marketplace.md (hunk 2 above)
Comment (abuccts) [reply docs/protocol/marketplace.md]: It's a recursive definition, "repository" is "OpenPAI repository".

File: docs/protocol/marketplace.md (hunk 3 above)
Comment (abuccts) [reply docs/protocol/marketplace.md]: I do have drawn a diagram in [previous version](https://github.com/Microsoft/pai/pull/2141/commits/1a6d42a9c69f9da0ee8b2075d883571766c0240b?short_path=8121237#diff-8121237fe21056489e7196f650c68b48). But to keep the design simple in the first stage and reserve minimum features to implement, removed it at present.

**Pull request closed without merging**